doc/source/dev/missing-bits.rst

```diff
@@ -0,0 +1,260 @@
+the preferred style is to use ``pytest.raises`` as a context manager, with
+the code that is supposed to raise the exception in the code block defined
+by the context manager.  The `match` keyword argument of ``pytest.raises``
+is given with enough of the expected error message attached to the exception
+to ensure that the expected exception is raised.
+
+For example, the function `scipy.stats.zmap` is supposed to raise a
+``ValueError`` if the input contains ``nan`` and ``nan_policy`` is ``"raise"``.
+A test for this is::
+
+    scores = np.array([1, 2, 3])
+    compare = np.array([-8, -3, 2, 7, 12, np.nan])
+    with pytest.raises(ValueError, match='input contains nan'):
+        stats.zmap(scores, compare, nan_policy='raise')
+
+The ``match`` argument ensures that the test doesn't pass by raising
+a ``ValueError`` that is not related to the input containing ``nan``.
+
+
+A parameter called lambda
+~~~~~~~~~~~~~~~~~~~~~~~~~
+There are functions where convention dictates that (ideally) a parameter
+should be called 'lambda'.  In Python, 'lambda' is a reserved word that
+cannot be used as a variable name.  If a new function or class has a
+parameter where a name that is similar to 'lambda' is really the best
+choice, the preferred name to use is [TBD: ``lam``, ``lmb``, ``lamb``,
+``lmbda``, ``lambda_``?].
+
+There has been a lack of consistency in the past:
+
+============================== =============
+Function or class              Variable name
+============================== =============
+`scipy.stats.boxcox`           ``lmbda``
+`scipy.stats.boxcox_llf`       ``lmb``
+`scipy.stats.yeojohnson`       ``lmbda``
+`scipy.stats.yeojohnson_llf`   ``lmb``
+`scipy.stats.chi2_contingency` ``lambda_``
+`scipy.stats.boltzmann`        ``lambda_``
+`scipy.stats.planck`           ``lambda_``
+`scipy.stats.tukeylambda`      ``lam``
+`scipy.signal.cspline1d`       ``lamb``
+`scipy.signal.qspline1d`       ``lamb``
+`scipy.signal.cspline2d`       ``lambda`` [*]
+`scipy.signal.qspline2d`       ``lambda`` [*]
+`scipy.signal.spline_filter`   ``lmbda``
+`scipy.special.tklmbda`        ``lmbda``
+`scipy.special.boxcox`         ``lmbda``
+`scipy.special.boxcox1p`       ``lmbda``
+`scipy.special.inv_boxcox`     ``lmbda``
+`scipy.special.inv_boxcox1p`   ``lmbda``
+============================== =============
+
+.. [*] This function accepts positional arguments only, but the docstring
+       documents the argument as `lambda`.
+
+
+Implicit string continuation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+When an implicit continuation is used to create a long literal string, the
+string is generally split at a space, and the space is included in the
+second part of the string.
+
+  Yes::
+
+      long_string = ("This is a very long string that demonstrates"
+                     " this guideline.")
+
+  No::
+
+      long_string = ("This is a very long string that demonstrates "
+                     "this guideline.")
```

Comment:
I don't think I've ever even considered blocking a PR for this differentiation. Using a `\` to continue will usually get me to say something though.

Reply:
Agreed, I'm happy to remove this completely.  I included it because it has come up in a couple NumPy PRs.

Reply:
Yes instead I would talk about `\` and suggest that we prefer one of the examples you showed.

Reply:
I have removed this guideline;  I don't think any of the SciPy devs care where the space goes, so we don't need it.

If it seems useful, we could add a note about "How to format a long string literal", that show our preference for using parentheses and implicit continuation instead of the backslash.  That could also be done in a follow-up PR.